Enhancement - Mejora - General - Permitir campos calculados.

ronaldchavezjortilles opened the pull request:
## Descripción del Cambio
Se agrego el botón de la funcionalidad de la creación de campos calculados. Luego, se trabajo que las columnas calculadas se puedan mantener en caso se ejecute el updatemodel. 

## Issue(s) resuelto(s)
- solves #322 

## Pruebas a realizar para validar el cambio

- Se realizo el updatemodel cuando no se crearon columnas calculadas (Funciona correctamente).
- Se realizo el updatemodel cuando el usuario creo un campo calculado (Funciona correctamente), es decir se actualiza todas las tablas y las columnas calculadas creadas, se mantienen. 
- Se verifico que las columnas calculadas creadas, sean funcionales para la creación de informes. 

## Información Adicional
Se hace el PR contra la rama integración porque es necesario que se haya integrado antes esa rama.

jortilles committed:
Dashboard Access Log

jortilles committed:
Analyze Query  Data

# Conflicts:
#	eda/eda_app/angular.json
#	eda/eda_app/proxy.conf.json

jortilles committed:
Update

jortilles committed:
Merge remote-tracking branch 'origin/onClickFilterFeature' into Integracion

# Conflicts:
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/eda-blank-panel.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/panel-charts/panel-chart.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/panel-utils/charts-config-utils.ts

jortilles committed:
Merge

jortilles committed:
consulta

ronaldchavezjortilles committed:
se hizo merge de la rama EDA#2165 sobre la rama integracion

ronaldchavezjortilles committed:
Modificación del z-index de dialog para ver el popup del Swal

ronaldchavezjortilles committed:
Agregado de pop up  de continuar y cancelar para ejecutar la tabla dataQuality

ronaldchavezjortilles committed:
Pop up de alerta para la tabla TablaQuality concluida

ronaldchavezjortilles committed:
Agregado de alerta de espera para el usuario cuando hace click abriendo los filtros

ronaldchavezjortilles committed:
etiqueta trans-unit agregada en el messages.en.xlf

jortilles committed:
logs del servidor

ronaldchavezjortilles committed:
Merge remote-tracking branch 'origin/Integracion' into integracion

ronaldchavezjortilles committed:
Agregado de componente logs, para logs y logs error

ronaldchavezjortilles committed:
Rutas logs y logs-error

ronaldchavezjortilles committed:
Merge branch 'integracion'

jortilles committed:
Updated Data Q

MarcCornejoJort committed:
table filters

MarcCornejoJort committed:
Merge branch 'onClickFilterFeature' into temporal

# Conflicts:
#	eda/eda_app/src/app/module/pages/dashboard/dashboard.component.ts

ronaldchavezjortilles committed:
Se convierte la función pivotTable en asincrona

ronaldchavezjortilles committed:
PivotTable asincrono

jortilles committed:
Merge remote-tracking branch 'origin/temporal'

jortilles committed:
comentados console log de mongo

jortilles committed:
update package

jortilles committed:
count distinct analyzie

jortilles committed:
root table es para eda2

ronaldchavezjortilles committed:
Tabla cruzada fix con panel de dragAndDrop

ronaldchavezjortilles committed:
Tabla cruzada verificada

jortilles committed:
role guard para los logs

jortilles committed:
logs para datasource creatos

jortilles committed:
volvemos a authgard

jortilles committed:
por defecto 5000

Por defecto se limitan las consultas a 5000 registros. si se quieren mas, que el usuario ponga un númeor más alto.

MarcCornejoJort committed:
addRelation with name + relationNameEmpty

ronaldchavezjortilles committed:
Corrección de issue dashboard.controller.ts y primary Key genérica para el treeMap

ronaldchavezjortilles committed:
Comentario de verificación

ronaldchavezjortilles committed:
Muestre los titulos del display_name

ronaldchavezjortilles committed:
etiquetas

ronaldchavezjortilles committed:
id commit

ronaldchavezjortilles committed:
Limpieza de código

MarcCornejoJort committed:
updateRelations done //TODO ==> separar update de create en relations

MarcCornejoJort committed:
updateLogic separada

ronaldchavezjortilles committed:
Arreglo de dimensionamientos de gráficos en el EBP

ronaldchavezjortilles committed:
Agregado de ? en la definición del KpiConfig

jortilles committed:
Revision consultas oracle generando el modelo

jortilles committed:
Merge branch 'master' of https://github.com/jortilles/EDA

jortilles committed:
Update Oracle Client

jortilles committed:
Data Q review count

MarcCornejoJort committed:
RelationNames Done

ronaldchavezjortilles committed:
bug repeticiones y porcentajes en tablas

MarcCornejoJort committed:
Bug tooltip arreglado

ronaldchavezjortilles committed:
Funcionalidad de filtros individuales en TreeTable completa

ronaldchavezjortilles committed:
Estilos de la celda del TreeTable

ronaldchavezjortilles committed:
Estilos completados del treeTable

MarcCornejoJort committed:
mapaBaseLayer + zoom

jortilles committed:
Merge branch 'customRelationsNames'

jortilles committed:
Merge remote-tracking branch 'origin/EDA#2183'

jortilles committed:
Translation typo

MarcCornejoJort committed:
bug dialog switch initial state

MarcCornejoJort committed:
bug coordenadas zoom guardado

jortilles committed:
Merge remote-tracking branch 'origin/customRelationsNames'

MarcCornejoJort committed:
provisional modification: coordmap drag depending on config file

MarcCornejoJort committed:
clean console

jortilles committed:
Merge remote-tracking branch 'origin/customRelationsNames'

ronaldchavezjortilles committed:
Agregado de el texto sin informar cuando el valor es null

ronaldchavezjortilles committed:
Comentario

ronaldchavezjortilles committed:
Valores nulos y vacios

ronaldchavezjortilles committed:
comentarios de valores vacios

ronaldchavezjortilles committed:
Aumento de padding

jortilles committed:
Docker update

jortilles committed:
docker

jortilles committed:
otro sudo

jortilles committed:
docker run

jortilles committed:
version

ronaldchavezjortilles committed:
Agregado de toString() en el mySql-builder.service.ts

ronaldchavezjortilles committed:
Merge branch 'master' of https://github.com/jortilles/EDA

ronaldchavezjortilles committed:
Agregado de comentario

ronaldchavezjortilles committed:
Agregado de límite - mostrar top para las fuentes de datos Excel

jortilles committed:
Corrección de error tipográfico.

jortilles committed:
txts updates

jortilles committed:
commit

jortilles committed:
customize data labels  for pyramid chart.js.plugin.toolptip

Commit:
added one click deploy

jortilles committed:
console.logs limpiados

jortilles committed:
Merge pull request #95 from Lasim/feat/one-click

added one click deploy

jortilles committed:
null es null

jortilles committed:
Merge branch 'master' of https://github.com/jortilles/EDA

jortilles committed:
Mongo DB Auth source

jortilles committed:
Comments 1

jortilles committed:
estilos SDA

ronaldchavezjortilles committed:
Preparación del componente para agregar la lógica que permite una DynamicTreeTable

ronaldchavezjortilles committed:
Dynamic Tree Table version 1 funcional

ronaldchavezjortilles committed:
Agregado de estilo a las cabeceras

ronaldchavezjortilles committed:
Control de valores

ronaldchavezjortilles committed:
Agregado de filtros de la tabla árbol dinámica

ronaldchavezjortilles committed:
limpieza de código

ronaldchavezjortilles committed:
Avance de filtros por usuario

jortilles committed:
oculta los informes públicos si no los puedo ver.

ronaldchavezjortilles committed:
creación del componente dialog para el treeTable

ronaldchavezjortilles committed:
Se abre la ventana app-tree-table-dialog

ronaldchavezjortilles committed:
Control del dialog recepcion del gráfico treeTable y emisión de 0 o 1 dependiendo de cancelar y confirmar

ronaldchavezjortilles committed:
agregado de comentario treeTableDialog

ronaldchavezjortilles committed:
Creación del modelo TreeTableConfig y llega al treeTableDialog - falta configurar el setConfig del charts-config-utils.ts

jortilles committed:
console log

jortilles committed:
ordenamiento de acces al logs

ronaldchavezjortilles committed:
Guardamos los cambios del config treeTable  realizados en la edición

ronaldchavezjortilles committed:
Funciona el cambio en las prueba iniciales

ronaldchavezjortilles committed:
Agregado de título a la edición de la tabla árbol

ronaldchavezjortilles committed:
Agregado de pickList al componente TreeTableDialog

ronaldchavezjortilles committed:
Funciones de intercambio de items

ronaldchavezjortilles committed:
Ocultar los botones del source y del target

ronaldchavezjortilles committed:
Finalización del treeTableDialog y limpieza de código

ronaldchavezjortilles committed:
El componente tree-table se reinicia al ejecutar desde el boton Ejecutar del Eda-Blank-Panel

ronaldchavezjortilles committed:
Colocado en negrita la cabezera del árbol

ronaldchavezjortilles committed:
Merge branch 'EDA#2206'

ronaldchavezjortilles committed:
Nombres del componente TreeTable desde el display_default

ronaldchavezjortilles committed:
Merge branch 'EDA#2206'

ronaldchavezjortilles committed:
Reducir el tamaño del input, cuando es mayor a 12 labels

ronaldchavezjortilles committed:
Merge branch 'EDA#2206'

ronaldchavezjortilles committed:
longitud del label para una tabla árbol básica

ronaldchavezjortilles committed:
Merge branch 'EDA#2206'

ronaldchavezjortilles committed:
Variado de tamaño del input a 80px de la tabla árbol

ronaldchavezjortilles committed:
Merge branch 'EDA#2206'

jortilles committed:
Comentarios a ingles

jortilles committed:
limpieza

jortilles committed:
Virgencita. Virgencita, que miedo me da. Merge branch 'master' of https://github.com/jortilles/EDA into integracion_eda_2.1.3

# Conflicts:
#	eda/SdaVersion.js
#	eda/eda_api/config/base_datamodel.json
#	eda/eda_api/lib/guards/updateModel-guard.ts
#	eda/eda_api/lib/module/dashboard/dashboard.controller.ts
#	eda/eda_api/lib/module/updateModel/models/users.ts
#	eda/eda_api/lib/module/updateModel/service/cleanModel.ts
#	eda/eda_api/lib/module/updateModel/service/push.Model.to.Mongo.ts
#	eda/eda_api/lib/module/updateModel/service/usersAndGroupsToMongo.ts
#	eda/eda_api/lib/module/updateModel/updateModel.controller.ts
#	eda/eda_api/lib/server.ts
#	eda/eda_api/lib/services/connection/db-systems/mysql-connection.ts
#	eda/eda_api/lib/services/custom/custom.ts
#	eda/eda_api/lib/services/query-builder/qb-systems/mySql-builder.service.ts
#	eda/eda_api/lib/services/query-builder/query-builder.service.ts
#	eda/eda_api/package.json
#	eda/eda_app/angular.json
#	eda/eda_app/package.json
#	eda/eda_app/src/app/module/components/components.module.ts
#	eda/eda_app/src/app/module/components/drag-drop/drag-drop.component.css
#	eda/eda_app/src/app/module/components/drag-drop/drag-drop.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/eda-blank-panel.component.html
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/eda-blank-panel.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/filter-dialog/filter-dialog.component.html
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/panel-charts/chart-configuration-models/table-config.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/panel-charts/panel-chart.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/panel-utils/charts-config-utils.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/panel-utils/query-utils.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/table-dialog/table-dialog.component.ts
#	eda/eda_app/src/app/module/components/eda-table/eda-table.component.ts
#	eda/eda_app/src/app/module/components/eda-table/eda-table.ts
#	eda/eda_app/src/app/module/pages/dashboard/dashboard.component.css
#	eda/eda_app/src/app/module/pages/dashboard/dashboard.component.html
#	eda/eda_app/src/app/module/pages/dashboard/dashboard.component.ts
#	eda/eda_app/src/app/module/pages/dashboard/global-filter/global-filter.component.html
#	eda/eda_app/src/app/module/pages/dashboard/global-filter/global-filter.component.ts
#	eda/eda_app/src/app/module/pages/data-sources/data-source-detail/sda-data-source-detail.component.html
#	eda/eda_app/src/app/module/pages/pages.module.ts
#	eda/eda_app/src/app/module/pages/pages.routes.ts
#	eda/eda_app/src/app/shared/components/create-dashboard/create-dashboard.component.ts
#	eda/eda_app/src/assets/sass/eda-styles/components/eda-blank-panel.css
#	eda/eda_app/src/locale/messages.ca.xlf
#	eda/eda_app/src/locale/messages.en.xlf
#	eda/eda_app/src/locale/messages.es.xlf
#	eda/eda_app/src/locale/messages.gl.xlf
#	eda/eda_app/src/locale/messages.pl.xlf
#	eda/eda_app/src/locale/stic_messages.ca.xlf
#	eda/eda_app/src/locale/stic_messages.en.xlf
#	eda/eda_app/src/locale/stic_messages.es.xlf
#	eda/eda_app/src/locale/stic_messages.gl.xlf

ronaldchavezjortilles committed:
oracledb en el packageJson

ronaldchavezjortilles committed:
merge EDA Frontend

ronaldchavezjortilles committed:
Merge remote-tracking branch 'origin/develop' into integracion_eda_2.1.3

integracion develop and or
# Conflicts:
#	eda/eda_api/lib/services/connection/db-systems/oracle-connection.ts
#	eda/eda_api/lib/services/query-builder/qb-systems/mySql-builder.service.ts
#	eda/eda_api/lib/services/query-builder/query-builder.service.ts
#	eda/eda_app/package.json
#	eda/eda_app/src/app/module/components/component.index.ts
#	eda/eda_app/src/app/module/components/components.module.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/eda-blank-panel.component.html
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/eda-blank-panel.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/panel-charts/panel-chart.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/panel-utils/query-utils.ts
#	eda/eda_app/src/app/module/components/eda-table/eda-table.ts
#	eda/eda_app/src/app/module/pages/dashboard/dashboard.component.ts
#	eda/eda_app/src/locale/messages.en.xlf
#	eda/eda_app/src/translate/messagesBK.xlf

ronaldchavezjortilles committed:
Migración funcional 1

ronaldchavezjortilles committed:
Filter-and-or guardado en servidor corregido

ronaldchavezjortilles committed:
eda-filter-and-or estilos

ronaldchavezjortilles committed:
Correcciones de errores para usuarios no administradores

ronaldchavezjortilles committed:
Usuarios no administradores, corrección de valores nulos

ronaldchavezjortilles committed:
Ojo de visualización de campos ocultos, ahora ya disponible

ronaldchavezjortilles committed:
Limpieza de console.logs

ronaldchavezjortilles committed:
eda-blank-panel y panel-interaction-utils

ronaldchavezjortilles committed:
Ajustes de SDA CUSTOM

ronaldchavezjortilles committed:
verificación de query-builder.service.ts

ronaldchavezjortilles committed:
Ultimos cambios de las variaciones del SDA CUSTOM

ronaldchavezjortilles committed:
modificación en la función onPanelAction en el dashboard.component.ts

ronaldchavezjortilles committed:
rootTable en undefined cuando se cancela el panel

ronaldchavezjortilles committed:
Corrección de función de descarga PDF

ronaldchavezjortilles committed:
Finalización de la implementación de Logs

ronaldchavezjortilles committed:
Adaptacion de modificaciones en EDA gráficos

ronaldchavezjortilles committed:
Agregado de funcion de filtrado con click en la tabla de tipo treetable

jortilles committed:
comentarios

jortilles committed:
log

jortilles committed:
log

jortilles committed:
connection

jortilles committed:
restaurado

jortilles committed:
restaurado

jortilles committed:
typo

jortilles committed:
paquete repetido

jortilles committed:
linea repetida

ronaldchavezjortilles committed:
Correción de botón de confirmación para las etiquetas

ronaldchavezjortilles committed:
Correción de caso particular para la TablaQuality en el back-end, solucionado

ronaldchavezjortilles committed:
Corrección de visibilidad de acción personalizada dependiendo de los permisos del usuario

ronaldchavezjortilles committed:
Cambio al inglés del módulo FuncionalidadURL a UrlFunctionality

ronaldchavezjortilles committed:
Cambio de nombre a CustomActionCall

ronaldchavezjortilles committed:
Verificación lista para informes comunes, parte 1

ronaldchavezjortilles committed:
Control de visualización de informes para usuarios No Administradores, para informes comunes e informes públicos

ronaldchavezjortilles committed:
Cambio de fuente en toda la applicación a fuente => Questrial

ronaldchavezjortilles committed:
control de visualización de tabla tipo árol

jortilles committed:
eliminación de comentarios

jortilles committed:
quitados console log

juanSTIC committed:
eliminar paquete repetido

juanSTIC committed:
Evitar mostrar etiquetas vacías

ronaldchavezjortilles committed:
información de no utilidad de tabla tipo árbol

ronaldchavezjortilles committed:
Actualización de texto informatico de tabla tipo árbol

ronaldchavezjortilles committed:
Corrección del texto tabla tipo árbol

ronaldchavezjortilles committed:
Agregado de descripción de kpi + barras

ronaldchavezjortilles committed:
descripción de KPI + gráficos de lineas

ronaldchavezjortilles committed:
Descripción de gráfico de area

ronaldchavezjortilles committed:
Corrección del KPI + gráfico de linea

ronaldchavezjortilles committed:
Corrección de KPI + Gráfico de area

ronaldchavezjortilles committed:
información del gráfico radar

ronaldchavezjortilles committed:
Correción de gráfico pirámide

ronaldchavezjortilles committed:
Ordenamiento alfabético de los tags

ronaldchavezjortilles committed:
Correción de issue de colores en el TreeMap

jortilles committed:
descripción del tree table

jortilles committed:
commit

ronaldchavezjortilles committed:
Desbloqueo del boton de los campos calculados

ronaldchavezjortilles committed:
Funcionalidad de manejo y control de campos calculados cuando realizamos el updatemodel de SinergiaDA

MarcCornejoJort committed:
arregloColor Blanco

ronaldchavezjortilles committed:
Correción del gráfico Tabla DataQuality

ronaldchavezjortilles committed:
Merge branch 'parallelIssue' into integracion_eda_2.1.3

ronaldchavezjortilles committed:
Corrección de etiquetas

ronaldchavezjortilles committed:
control de visualización de table DataQuality para usuarios sin permisos

ronaldchavezjortilles committed:
Los informes públicos ahora ya no permiten la opción de Guardar Como

MarcCornejoJort committed:
securityTable fix

MarcCornejoJort committed:
Merge branch 'integracion_eda_2.1.3' of https://github.com/SinergiaTIC/SinergiaDA into integracion_eda_2.1.3

ronaldchavezjortilles committed:
Corrección de issue, de visibilidad de opciones del sidebar para un informe público

ronaldchavezjortilles committed:
Ordenamiento de las etiquetas del inicio de SinergiaDA

ronaldchavezjortilles committed:
joins agregado para los filtros de tipo fecha

PaulaaSTIC reviewed (changes requested):
- Está bloqueada la edición de campos calculados.
- Al añadir una columna, solo aparece el asterisco (*) de campo obligatorio en nombre, pero si no pongo descripción me salta igualmente el mensaje de campo requerido.

![Prueba_cc](https://github.com/user-attachments/assets/a78297e3-d788-4213-aa64-ff61c008e938)

juanSTIC committed:
Mejora en la visualización de etiquetas en el dashboard, permitiendo múltiples etiquetas por informe.

ronaldchavezjortilles committed:
Queda desbloqueado el panel cuando es un campo calculado

ronaldchavezjortilles committed:
Agregado de asterisco

ronaldchavezjortilles commented:
Se realizo las dos modificaciones pendientes, listo para su revisión.

PaulaaSTIC committed:
Compatibility of old report tags

ronaldchavezjortilles committed:
Correción de Tabla DataQuality

juanSTIC commented:
Después de reunión (2/9/25) se acuerda que:
- La interfaz de los campos calculados de cualquier tipo (incluidos los numéricos), no exigirán utilizar ningún tipo de agregación ya que de esta manera se evita tener que usar necesariamente los campos calculados en el HAVING si se usan para filtrar.
- Al usar los campo calculados en el editor de consultas, estarán disponibles (de manera automática)  las agregaciones propias de cada tipo de campo:
   - texto: count,count_distinct,none
   - date : count,count_distinct,none
   - numeric: sum,avg,max,min,count,count_distinct,none
- El usuario no debe poder elegir las agregaciones en cada campo calculado, sino que se asignan de manera automática según lo anterior

jortilles committed:
Eliminación de filtros por agregaciones

ronaldchavezjortilles committed:
Nueva implementación en el modelo de datos de agregaciones tipo text, date, numeric y coordinate

ronaldchavezjortilles committed:
Agrego de elementos en el objeto filtro: SQLexpression y computed_column

ronaldchavezjortilles committed:
Limpieza de console.logs

juanSTIC committed:
Mejora en la normalización de etiquetas y limpieza de código en HomeSdaComponent

ronaldchavezjortilles committed:
Usuario no admin, podra hacer guardar como

ronaldchavezjortilles committed:
tratado de dashboards en los grupos

ronaldchavezjortilles committed:
Limpieza de código

ronaldchavezjortilles committed:
refactory de función getSeparedColumns para las columnas calculadas

ronaldchavezjortilles committed:
filtros de columna en el panel

ronaldchavezjortilles committed:
Filtros en los paneles que no están seleccionados en las columnas

juanSTIC committed:
Mejora en la traducción y limpieza de espacios en el archivo de localización

PaulaaSTIC committed:
Merge branch 'develop' into integracion_eda_2.1.3

PaulaaSTIC committed:
Merge branch 'integracion_eda_2.1.3' into develop

PaulaaSTIC committed:
Merge branch 'integracion_eda_2.1.3' into SDA#322

PaulaaSTIC committed:
Merge branch 'develop' into SDA#322

ronaldchavezjortilles committed:
Control de filtros avanzados y filtros globales

ronaldchavezjortilles committed:
Internacionalización de campos calculados

ronaldchavezjortilles committed:
Agregado de filtros de campos calculados en la seccion where

ronaldchavezjortilles committed:
Placeholder

jortilles committed:
cadena original corregida

ronaldchavezjortilles committed:
Merge remote-tracking branch 'origin/develop' into SDA#322

# Conflicts:
#	eda/eda_api/lib/services/query-builder/qb-systems/mySql-builder.service.ts
#	eda/eda_app/src/app/module/components/eda-filter-and-or/eda-filter-and-or.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/column-dialog/column-dialog.component.ts
#	eda/eda_app/src/app/module/components/eda-panels/eda-blank-panel/filter-dialog/filter-dialog.component.ts
#	eda/eda_app/src/app/services/api/global-filters.service.ts
#	eda/eda_app/src/app/services/utils/column-utils.service.ts

jortilles committed:
mergeado develop

Al mergear develop se ha mergeado también el desarrollo de coger los código para los filtros. Eso hacía que hubiera que gestionar también la casuistica de las expresiones SQL para los filtros en el nuevo formado, id +  descripción.

jortilles commented:
Se ha mergeado develop y se ha visto que había que tener en cuenta las expresiones calculadas en los filtros de tipo id  +  descripción. 

Por otro lado. Tal y cómo se detectó. Es necesario ser cuidadoso a la hora de generar las expresiones SQL y  poner expresiones completas. No obviando nombres de tabla. Es decir decir tabla.campoA + tabla.campoB y no usar abreviaturas del tipo campoA+CampoB porque hay muchas tablas con campos con nombres iguales que producirán  errores SQL

juanSTIC reviewed (changes requested):
La interfaz permite establecer una expresión como `AVG(duration)` en un campo calculado y la da por buena, sin embargo siempre fallará al utilizarla, por lo que habría que impedir que se pueda registrar ninguna agregación en los campos calculados.
<img width="1622" height="838" alt="image" src="https://github.com/user-attachments/assets/e4bcdfc3-176f-4a9e-81c2-d02132882092" />
Diría que se puede asegurar, evitando que las siguientes palabras aparezcan en la expresión (tanto en minúsculas como en mayúsculas):

```
[
  "AVG",
  "BIT_AND",
  "BIT_OR",
  "BIT_XOR",
  "COUNT",
  "GROUP_CONCAT",
  "JSON_ARRAYAGG",
  "JSON_OBJECTAGG",
  "MAX",
  "MIN",
  "STD",
  "STDDEV",
  "SUM",
  "VAR_POP",
  "VAR_SAMP",
  "VARIANCE",
  "CUME_DIST",
  "DENSE_RANK",
  "FIRST_VALUE",
  "LAG",
  "LAST_VALUE",
  "LEAD",
  "NTH_VALUE",
  "NTILE",
  "PERCENT_RANK",
  "RANK",
  "ROW_NUMBER"
]
```

ronaldchavezjortilles committed:
No permitir el añadido de agregaciones en el los paneles de los campos calculados

juanSTIC reviewed (changes requested):
El sistema permite crear dos o más campos calculados con el mismo nombre en la misma tabla, lo cual lleva siempre a error. Antes de crearse el campo calculado debe comprobarse que no exista.

ronaldchavezjortilles committed:
lista de agregaciones actualizada y la app no permite crear campos calculados con el mismo nombre

jortilles committed:
comments

jortilles committed:
Control

de mayusculas y minusculas y espacios en blanco en los nombres

jortilles commented:
Ya esta corregido y ya se puede volver a evaluar. Hemos controlado también espacios en blanco y mayusculas por si acaso.

PaulaaSTIC reviewed (changes requested):
Si se utiliza alguna de las agregaciones bloqueadas (AVG, SUM, etc.), al pulsar el botón Comprobar expresión aparece el mensaje: “No se pueden incluir agregaciones en la consulta (DISTINCT, SUM, MAX, MIN, etc.)”. Sin embargo, luego nos permite guardar el modelo sin problema. Además, pulsar el botón Comprobar expresión no es obligatorio, por lo que podríamos haber guardado el modelo sin saber que la expresión no es válida.

De la misma forma, Comprobar expresión realmente no verifica que haya concordancia entre el resultado de la expresión y el tipo seleccionado, lo que permite indicar que valores que no son de tipo fecha o texto lo sean.

También se permite guardar un campo calculado sin expresión, y al usarlo en un panel se produce un error: “Querying database”.

jortilles commented:
> Si se utiliza alguna de las agregaciones bloqueadas (AVG, SUM, etc.), al pulsar el botón Comprobar expresión aparece el mensaje: “No se pueden incluir agregaciones en la consulta (DISTINCT, SUM, MAX, MIN, etc.)”. Sin embargo, luego nos permite guardar el modelo sin problema. Además, pulsar el botón Comprobar expresión no es obligatorio, por lo que podríamos haber guardado el modelo sin saber que la expresión no es válida.
> 
> De la misma forma, Comprobar expresión realmente no verifica que haya concordancia entre el resultado de la expresión y el tipo seleccionado, lo que permite indicar que valores que no son de tipo fecha o texto lo sean.
> 
> También se permite guardar un campo calculado sin expresión, y al usarlo en un panel se produce un error: “Querying database”.


Como resultado de la conversación mantenida a raíz de este comentario se propone:

1. mover toda la interacción de los campos calculados  a la ventana modal. En esta ventana modal. Hay que validar el campo para que entre en el modelo. (Expresión sql, nombres repetidos, etc)
2. El formulario del campo calculado en el modelo será de sólo lectura con un botón de editar campo calculado

Esto supone un tiempo de desarrollo de unas 6 horas en desarrollar el código y validarlo. Lo que supone una ligera variación del alcance a 53 horas.

ronaldchavezjortilles committed:
Primera parte de de campos calculados

ronaldchavezjortilles committed:
Avance 2 de control de información recibida en sobre los campos calculados desde su creación

ronaldchavezjortilles committed:
Verificación de un query correcta para el campo calculado....

ronaldchavezjortilles committed:
Creación del dialog para la edición de campos calculados

ronaldchavezjortilles committed:
dialog creado y validación del formulario

ronaldchavezjortilles committed:
Edición de campo calculado terminado

ronaldchavezjortilles committed:
Control de mismo nombre en la creación y edición de campos calculados

ronaldchavezjortilles committed:
control de aggregaciones que no permitidas para la creación y edición de campos calculados

ronaldchavezjortilles committed:
Cambio de textos

ronaldchavezjortilles committed:
Limpieza y finalización del código

jortilles committed:
mensajes

jortilles committed:
texto agregado a 0 decimales

ronaldchavezjortilles committed:
control de nulos para los campos diferentes de numeric

jortilles commented:
Se ha desarrollado la nueva forma de generar los campos calculados y está lista para validar.

juanSTIC reviewed (changes requested):
Al eliminar un campo calculado no se pide confirmación y debería.

juanSTIC reviewed (changes requested):
Al editar un campo calculado **que ya existe**, debe de mostrarse una alerta con el mensaje "Al modificar  un campo calculado ya existente los paneles que lo utilizan  no se actualizan automáticamente, es necesario entra en la configuración de cada uno de ellos, pulsar en Ejecutar y guardar el informe"

ronaldchavezjortilles committed:
Mensajes de alerta para los campos calculados

jortilles commented:
Ya se pueden validar los cambios

jortilles committed:
texto del issue

juanSTIC committed:
Icono condicionado al tipo de columna

juanSTIC reviewed (changes requested):
Los formatos de fecha no aplican sobre campos calculados cuando se utilizan DATE_ADD o DATE_SUB (y probablemente con otras funciones que devuelven un campo tipo fecha).

Cuando se crea un campo calculado de tipo fecha, por ejemplo con la función `DATE_SUB(date_entered, INTERVAL 1 MONTH)` el resultado se muestra como _2025-05-12 13:01:16.000000_ y ya no se pueden aplicar los formatos predefinidos que hay en la pantalla de edición de campos tipo fecha.

También sucede que si en un campo calculado de tipo fecha, se pone como expresión el nombre de un campo tipo fecha (por ejemplo `date_entered`) lo caual es una expresión válida, el campo calculado se muestra con un formato diferente al original, mostrandose en formato Y-m-d H-M-s y tampoco es posible aplicar sobre el campo los formatos predefinidos.

juanSTIC committed:
Agregar mensajes informativos para columnas calculadas sin opciones de formato disponibles

juanSTIC committed:
Agregar mensajes informativos para columnas calculadas sin opciones de formato disponibles

juanSTIC committed:
Merge branch 'SDA#322' of https://github.com/SinergiaTIC/SinergiaDA into SDA#322

juanSTIC committed:
Merge branch 'SDA#322' of https://github.com/SinergiaTIC/SinergiaDA into SDA#322

jalbaiges committed:
Improving translation

jalbaiges committed:
Improving translation

jalbaiges committed:
Improving translation

jalbaiges reviewed (commented):
Cadenas revisadas.

jortilles commented:
PR https://github.com/SinergiaTIC/SinergiaDA/pull/375

PaulaaSTIC reviewed (approved):
(A)probado

juanSTIC reviewed (approved):
Visto y probado
